lib/gooddata/models/client.rb

```diff
@@ -39,7 +39,9 @@ def [](id, opts = {})
         fail ArgumentError, 'No client specified' if client.nil?
 
         if id == :all
-          tenants_uri = domain.segments_uri + "/clients?segment=#{CGI.escape(segment.segment_id)}"
+          data_product = segment.data_product
+          tenants_uri = GoodData::DataProduct::ONE_DATA_PRODUCT_PATH % { domain_name: domain.name, id: data_product.data_product_id }
```

Comment:
Refers to 'domain' more than self (maybe move it to another class?) (FeatureEnvy)